pkg/e2e/machinehealthcheck/machinehealthcheck.go

```diff
@@ -57,9 +58,26 @@ var _ = Describe("[Feature:MachineHealthCheck] MachineHealthCheck controller", f
 		client, err = e2e.LoadClient()
 		Expect(err).ToNot(HaveOccurred())
 
-		// TODO: enable once https://github.com/openshift/cluster-api-actuator-pkg/pull/61 is fixed
-		glog.V(2).Info("Skipping machine health checking test")
-		Skip("Skipping machine health checking test")
+		isKubemarkProvider, err := e2e.IsKubemarkProvider(client)
+		Expect(err).ToNot(HaveOccurred())
+		// TODO: remove once we can create or update kubemark machines
+		// that will give use possibility to make this test work
+		if isKubemarkProvider {
+			glog.V(2).Info("Can not run this tests with the 'KubeMark' provider")
+			Skip("Can not run this tests with the 'KubeMark' provider")
+		}
+
+		err = e2e.CreateTechPreviewFeatureGate()
+		Expect(err).ToNot(HaveOccurred())
+
```

Comment:
why are we deleting the deployment here?

Reply:
to make machine-api-operator reconcile happen faster and to add machine-health-check container

Reply:
shouldn't it be enough to `CreateTechPreviewFeatureGate`?

Reply:
I think it should, but will make harder to monitor when the updated deployment is ready and can introduce some flakiness, so I prefer to explicitly delete the deployment

Reply:
why you need to check when the deployment is ready? when the new controller comes up should see the created hc resource, right? Why flakiness? The tests should reproduce the expected user interaction, if some of this does not work there's a bug

Reply:
done